### PR TITLE
cocktail sort in c

### PR DESCRIPTION
**Pull request by TechnoPhasePRO**
**Fixes issue:**#4644

**Changes:**

Added code for cocktail sort in c programming

### CODE REVIEW

File: code/sorting/src/cocktail_sort/cocktail_sort.c [hunk 1]
```diff
@@ -0,0 +1,65 @@
+#include <stdio.h>
+#include <conio.h>
```
Comment (arnavb) [inline code/sorting/src/cocktail_sort/cocktail_sort.c]: Don't use this header. Reformat both your files to use clang-format with 4 spaces.

File: code/sorting/src/cocktail_sort/cocktail_sort.c [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            }
+            /* after every sorting, largest element is at the last index and value of p is set to 1 */
+        } 
+        if (!p)
+        /* if value is not 1, it means the array is sorted already , so the control is sent out of while loop */
+            break;
+        else {
+        /* value of last is decreased by 1 as largest element need not to be sorted */
+            p = 0;
+            last= last-1;
```
Comment (arnavb) [inline code/sorting/src/cocktail_sort/cocktail_sort.c]: Use compound assignment whenever possible.

File: code/sorting/src/cocktail_sort/cocktail_sort.c [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+/* CocktailSort function is the function used to sort the array using cocktail algorithm. */
+/* Cocktailsort is similar to the bubble sort but the complexity is less comparatively. */
+void CocktailSort(int l[], int n)  
+{   
+    /* argument passed is the list of the array to be sorted and the length of the array */
+    /* first is used to represent the starting index of the array
+     * last is used to represent the last index of the array
+     * p is the variable used to check if the array contains sorted elements after sorting process
+     */
+    int first = 0, k, last, temp = 0, p=1;
```
Comment (arnavb) [inline code/sorting/src/cocktail_sort/cocktail_sort.c]: Declare these variables where they are first used, not at the top of the function.

File: code/sorting/src/cocktail_sort/cocktail_sort.cpp [hunk 4]
```diff
@@ -0,0 +1,65 @@
+#include <stdio.h>
```
Comment (arnavb) [inline code/sorting/src/cocktail_sort/cocktail_sort.cpp]: Don't use either of these headers in C++. Use `<iostream>`

File: code/sorting/src/cocktail_sort/cocktail_sort.cpp [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+                temp = l[i];
+                l[i] = l[i+1];
+                l[i+1] = temp;
+                p = 1; 
+            } 
+        }
+        /* the smallest element is placed at the front of the array, so the value of first variable is increased by 1 */
+        first = first+1;
+    }
+    for (int i = 0; i < n; i++)
```
Comment (arnavb) [inline code/sorting/src/cocktail_sort/cocktail_sort.cpp]: Pe-increment  in  loops.

**Comment by TechnoPhasePRO** [comment]
@arnavb  the program was in c language but extension was as .cpp. although i still used c extension but in other code and made PR on it.

File: code/sorting/src/cocktail_sort/cocktail_sort.c (hunk 1 above)
Comment (TechnoPhasePRO) [reply code/sorting/src/cocktail_sort/cocktail_sort.c]: can you please tell me what i need to do exactly?

**Comment by TechnoPhasePRO** [comment]
@arnavb sir, please look at the modified code of cocktail sort in c.

File: code/sorting/src/cocktail_sort/cocktail_sort.c [hunk 6]
```diff
@@ -0,0 +1,67 @@
+#include "stdio.h"
```
Comment (arnavb) [inline code/sorting/src/cocktail_sort/cocktail_sort.c]: Use brackets. Don't use conio.h

File: code/sorting/src/cocktail_sort/cocktail_sort.c [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+#include "conio.h"
+
+/* CocktailSort function is the function used to sort the array using cocktail algorithm. */
+/* Cocktailsort is similar to the bubble sort but the complexity is less comparatively. */
+void CocktailSort(int l[], int n)
```
Comment (arnavb) [inline code/sorting/src/cocktail_sort/cocktail_sort.c]: lowerCamelCase.

File: code/sorting/src/cocktail_sort/cocktail_sort.c [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+void CocktailSort(int l[], int n)  
+{   
+    /* argument passed is the list of the array to be sorted and the length of the array */
+    /* first is used to represent the starting index of the array
+     * last is used to represent the last index of the array
+     * p is the variable used to check if the array contains sorted elements after sorting process
+     */
+    int first = 0;
+    int last = n-1;
+    int temp = 0;
```
Comment (arnavb) [inline code/sorting/src/cocktail_sort/cocktail_sort.c]: Reformat this file using clang format 4 spaces.

File: code/sorting/src/cocktail_sort/cocktail_sort.c [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+            if (l[i] > l[i + 1]) 
+            { 
+                temp = l[i];
+                l[i] = l[i+1];
+                l[i+1] = temp;
+                p = 1; 
+            } 
+        }
+        /* the smallest element is placed at the front of the array, so the value of first variable is increased by 1 */
+        first+=1;
```
Comment (arnavb) [inline code/sorting/src/cocktail_sort/cocktail_sort.c]:
```suggestion
        first++;
```

File: code/sorting/src/cocktail_sort/cocktail_sort.c (hunk 8 above)
Comment (TechnoPhasePRO) [reply code/sorting/src/cocktail_sort/cocktail_sort.c]: @arnavb  sir,would you please elaborate what exactly is to be done?

Comment (TechnoPhasePRO) [reply code/sorting/src/cocktail_sort/cocktail_sort.c]: sir, now it's indented to 4 spaces. please look at it. @arnavb

**Comment by TechnoPhasePRO** [comment]
@arnavb  sir, new modified code has been added and indented . please look at it.

File: code/sorting/src/cocktail_sort/cocktail_sort.c [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+        /* the smallest element is placed at the front of the array, so the value of first variable is increased by 1 */
+        first++;
+    }
+    for (int i = 0; i < n; ++i) 
+    /* for loop is used to display the sorted array elements */
+        printf("%d ", l[i]); 
+    printf("\n");
+}
+
+void main()
```
Comment (arnavb) [inline code/sorting/src/cocktail_sort/cocktail_sort.c]: int.

**Comment by arnavb** [comment]
@TechnoPhasePRO You can remove your C++ code because I am going to merge C++ cocktail sort from #4889

**Pull request closed without merging**